src/common/VirtualTerminal.cpp

```diff
@@ -128,17 +147,28 @@ namespace SDDM {
                 close(fd);
             });
 
+#ifdef __FreeBSD__
+            int vtActive = 0;
+            if (ioctl(fd, VT_GETACTIVE, &vtActive) < 0) {
+                qCritical() << "Failed to get current VT:" << strerror(errno);
+                close(fd);
+                return -1;
+            }
+#else
```

Comment:
Some fly-by comments, from a non-sddm stakeholder. HTH o/

This seems like a good candidate for a FreeBSD/Linux helper, just like the `getVtPath()` above.
The extra `close(fd)` should not be needed, since we have the `qScopeGuard { close(fd); }` just above.